src/promptflow-core/promptflow/_utils/multimedia_utils.py

```diff
@@ -2,278 +2,497 @@
+
+    @staticmethod
+    def create_image_from_url(url: str, mime_type: str = None):
+        response = requests.get(url)
+        if response.status_code == 200:
+            return Image(response.content, mime_type=mime_type, source_url=url)
+        else:
+            raise InvalidImageInput(
+                message_format="Failed to fetch image from URL: {url}. Error code: {error_code}. "
+                "Error message: {error_message}.",
+                target=ErrorTarget.EXECUTOR,
+                url=url,
+                error_code=response.status_code,
+                error_message=response.text,
+            )
 
-def _is_base64(value: str):
-    base64_regex = re.compile(r"^([A-Za-z0-9+/]{4})*(([A-Za-z0-9+/]{2})*(==|[A-Za-z0-9+/]=)?)?$")
-    if re.match(base64_regex, value):
-        return True
-    return False
-
-
-def _create_image_from_file(f: Path, mime_type: str = None):
-    with open(f, "rb") as fin:
-        return Image(fin.read(), mime_type=mime_type)
-
-
-def _create_image_from_base64(base64_str: str, mime_type: str = None):
-    image_bytes = base64.b64decode(base64_str)
-    return Image(image_bytes, mime_type=mime_type)
-
+    @staticmethod
+    def create_image_from_string(value: str):
+        if ImageProcessor.is_base64(value):
+            return ImageProcessor.create_image_from_base64(value)
+        elif ImageProcessor.is_url(value):
+            return ImageProcessor.create_image_from_url(value)
+        else:
+            return ImageProcessor.create_image_from_file(Path(value))
+
+
+class TextProcessor:
+    @staticmethod
+    def is_text_dict(text_dict: dict):
+        if len(text_dict) != 2:
+            return False
+        if "type" not in text_dict:
+            return False
+        if text_dict["type"] == "text" and "text" in text_dict:
+            text = text_dict["text"]
+            if isinstance(text, str):
+                return True
+            elif isinstance(text, dict):
+                if "value" in text and isinstance(text["value"], str):
+                    return True
+        return False
 
-def _create_image_from_url(url: str, mime_type: str = None):
-    response = requests.get(url)
-    if response.status_code == 200:
-        return Image(response.content, mime_type=mime_type, source_url=url)
-    else:
-        raise InvalidImageInput(
-            message_format="Failed to fetch image from URL: {url}. Error code: {error_code}. "
-            "Error message: {error_message}.",
-            target=ErrorTarget.EXECUTOR,
-            url=url,
-            error_code=response.status_code,
-            error_message=response.text,
+    @staticmethod
+    def create_text_from_dict(text_dict: any):
+        return Text.deserialize(text_dict)
+
+
+class MultimediaProcessor(ABC):
+    @staticmethod
+    def create(message_format_type: str = MessageFormatType.BASIC):
+        if (
+            not message_format_type
+            or message_format_type == ""
```

Comment:
`not message_format_type` contains `message_format_type == ""`, we can remove the later one.

Reply:
Removed empty value check.